### DISCUSSION TITLE
How to provide "Event Type Schema"?

### DISCUSSION THREAD

**Discussion by adimyth** [svix/svix-webhooks#581] [Q&A] [answered]
In the create event api, how do we provide "event type schema"?

**DOCS**
The [documentation](https://docs.svix.com/tutorials/event-type-schema) isn't clear about how do we translate a given example to corresponding json schema. It would have been nice to provide the json schema for the below example & explain how to do the conversion -

![](https://docs.svix.com/assets/images/schema-editor-basic-57bcf14b05162c19ba48b578d23d8bdc.png)

**REDOC**
The default example provided [here](https://api.svix.com/docs#operation/create_event_type_api_v1_event_type__post) isn't very helpful either. Sample request body looks like -
```json
{
  "description": "A user has signed up",
  "schemas": {
    "1": {
      "title": "Invoice Paid Event",
      "description": "An invoice was paid by a user",
      "type": "object",
      "properties": {
        "invoiceId": {
          "description": "The invoice id",
          "type": "string"
        },
        "userId": {
          "description": "The user id",
          "type": "string"
        }
      },
      "required": [
        "invoiceId",
        "userId"
      ]
    }
  },
  "archived": false,
  "name": "user.signup"
}
```
1. Should we expect `invoiceID` & `userID` in request body when the webhook is called for `user.signup` event? I created a very simple webhook endpoint, it returned 422 (unprocessable entity)
2. What is the meaning of `required`?
3. What are the possible values of `type`? Do we have to make a distinction between float16 & float32 for example

I would be happy to provide my use case if required & can help improve the docs

**Comment by tasn** [comment, marked as answer]
It's just a standard draft 7 json schema. You can use the visual editor (like in the screenshot) or click on the code tab to see the corresponding generated code.
You fully control the schema to whatever you want, it's a definition of what the content of the event would be. For Svix it doesn't matter how you design your schema. It's up to you.

So the answer to all of your questions is: check out json schema draft 7. :)

Does this make sense?

**Reply by adimyth** [reply]
Yeah, it makes sense. I just wondered if it would have been nice to provide a complete example in the docs & in the redoc as well, that's all

**Reply by tasn** [reply]
Yeah, it's probably a good idea.
